tests/flox-init.bats

```diff
@@ -0,0 +1,232 @@
+#! /usr/bin/env bats
+# -*- mode: bats; -*-
+# ============================================================================ #
+#
+# Test `flox init' sub-command.
+#
+# This sub-command's `--template' option takes an "installable" URI as its
+# argument, and we focus on auditing various URI edge cases here.
+#
+# NOTE: the URIs accepted by `flox init --template <URI>' do not accept an
+# extended output spec ( `^out,bin,dev' suffix ).
+#
+#
+# ---------------------------------------------------------------------------- #
+
+load test_support.bash;
+
+# bats file_tags=uri, init, uri:project
+
+
+# ---------------------------------------------------------------------------- #
+
+setup_file() {
+  common_file_setup;
+
+  # Suppresses warning messages that clutter backtraces.
+  git config --global init.defaultBranch main;
+
+  export _floxpkgs_rev="2c75b96bc3e8c78b516b1fc44dbf95deae6affca";
+
+  # Create an alias for testing indirects
+  $FLOX_CLI nix registry add floxpkgs-alias github:flox/floxpkgs;
+}
+
+
+# ---------------------------------------------------------------------------- #
+
+# Helpers for project based tests.
+
+project_setup() {
+  export PROJECT_DIR="${BATS_TEST_TMPDIR?}/project-${BATS_TEST_NUMBER?}";
+  rm -rf "$PROJECT_DIR";
+  mkdir -p "$PROJECT_DIR";
+  pushd "$PROJECT_DIR" >/dev/null||return;
```

Comment:
Question for my understanding: what purpose does `>/dev/null||return` serve?

Reply:
It causes the function to immediate return a failure code if the directory we are trying to change to does not exist.